api/util.py

```diff
@@ -171,6 +171,12 @@ def send_json_http_exception(response, message, code):
     response.headers['Content-Type'] = 'application/json; charset=utf-8'
     response.write(content)
 
+def send_custom_json_http_exception(response, errors, code):
+    response.set_status(code)
+    content = json.dumps({errors})
+    response.headers['Content-Type'] = 'application/json; charset=utf-8'
+    response.write(content)
```

Comment:
Any way we could constrain this to forcibly overlap with the above struct, `send_json_http_exception`?

I ask because this would add a lot of complexity to consuming the API from a static language. Right now, I can deserialize API errors into a struct and get useful things out. This change would mean errors are just an arbitrary json object :pensive: 

How about adding `custom=None` to the existing function, and I can consider that a wild west map?

```go
type ApiError struct {
	Message    string `json:"message"`
	StatusCode int    `json:"status_code"`
	
	// I would add this
	Custom map[string]interface `json:"custom"`
}
```